### ISSUE TITLE
About anthropometry

### ISSUE THREAD

**Issue by websocketing** [vladmandic/human#152]
I'm glad to see this project. I wonder if this project supports anthropometry? I need to measure the human parts of the generated manikin. I hope I can get your reply and information.

**Comment by websocketing** [comment]
@vladmandic

**Comment by vladmandic** [comment]
no, it doesn't support **anthropometry** - but that is a really interesting topic - i'll do some research on it  
and if you have a working example, let me know

**Comment by vladmandic** [comment]
from the top of my head...

given that we don't know distance or reference scale to start with, we need to start with either:

- focal length of a camera: can be user defined  
- reference object: item of a known size included in the image  

if focal length is well-known, its a decent start  

if relying on a refence object, there is issue with 3d depth:  
is reference object closer or further away from measured?  
could partially be solved with 3d reconstruction models, but that is pretty low precision  
only "real" way to do it would be using spectroscopic cameras so 3d reconstruction is precise  

without that, we simply don't know if we're looking at a doll in a model house or a giant on a hill  

and in both cases, we still have issues with  

- perspective distorsions: there are methods that measure distorsions,  
but all of them rely on some reference lines (not just points) - detecting horizont, floor, etc.
and again, issue with precision unless reference lines are well-defined (e.g. chess board as a wall)

- barrel distorsions: could be partially accounted for if camera field of view is well-known,  
otherwise this again becomes a major issue and only way of correcting distorsions are detection of curvature on what should be perfect lines  
but if relying on line curvature, that brings us back to 3d depth issue - is the detected line at the same depth as measured object?

so all-in-all, i don't think anthropometry would be any good without having a well-defined camera parameters to start with  
which fortunately in some cases can be extracted from the image exif properties for images, but it's missing for video feeds  

but not all cameras with same params have same level of distorsion - that depends on the actual glass elements
so most distorsion correction software works with a built-in lens distorsion database and still only corrects barrel distorsions  
while perspective distorsions are hit-or-miss (just look at photoshop or lightroom)

all-in-all, i could add some for-fun functions that do different measurements, but only if 
a) user provides camera parameters in the config
b) there are no significant perspective distorsions (meaning camera angle is straight-on)

let me know your thoughts

**Comment by vladmandic** [comment]
closing as it cannot be implemented without user providing a lot of information manually
